paywall/src/data-iframe/Mailbox.ts

```diff
@@ -316,6 +320,33 @@ export default class Mailbox {
     }
   }
 
+  /**
+   * When we receive PostMessages.SET_USER_METADATA, simply forward it
+   * to the handler
+   */
+  setUserMetadata = async ({
+    lockAddress,
+    metadata,
+  }: {
+    lockAddress: string
+    metadata: UserMetadata
+  }) => {
+    if (!this.handler) {
+      this.emitError(
+        new Error('blockchain handler not instantiated, cannot set metadata')
+      )
+      return
+    }
+
+    try {
+      await this.handler.setUserMetadata(lockAddress, metadata)
+      // success, no error returned
+      this.postMessage(PostMessages.SET_USER_METADATA_SUCCESS, undefined)
```

Comment:
The success branch has its own message. Typically the BlockchainHandler doesn't return successes, because they are eventually reflected in the data it emits. This use is different, because it's purely a side-effectful method that doesn't change the chain data we return.